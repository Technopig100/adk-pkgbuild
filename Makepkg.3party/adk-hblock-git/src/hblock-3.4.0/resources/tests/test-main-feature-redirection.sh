#!/bin/sh

# Author:     Héctor Molinero Fernández <hector@molinero.dev>
# License:    MIT, https://opensource.org/licenses/MIT
# Repository: https://github.com/hectorm/hblock

set -eu
export LC_ALL='C'

SCRIPT_DIR="$(CDPATH='' cd -- "$(dirname -- "${0:?}")" && pwd -P)"

# shellcheck disable=SC1091
. "${SCRIPT_DIR:?}"/env.sh

main() {
	export HBLOCK_SOURCES="file://${SCRIPT_DIR:?}/test-domains-source.txt"
	export HBLOCK_REDIRECTION='0.0.0.0'

	printf 'Test - Main - Redirection: "-R" short option\n'
	actual="$(runInTestShell "${SCRIPT_DIR:?}/../../hblock" -qO- -R '::1')"
	expected="$(cat -- "${0%.sh}".out)"
	if ! assertEquals "${actual?}" "${expected?}"; then
		exit 1
	fi

	printf 'Test - Main - Redirection: "--redirection" long option\n'
	actual="$(runInTestShell "${SCRIPT_DIR:?}/../../hblock" -qO- --redirection='::1')"
	expected="$(cat -- "${0%.sh}".out)"
	if ! assertEquals "${actual?}" "${expected?}"; then
		exit 1
	fi

	printf 'Test - Main - Redirection: "HBLOCK_REDIRECTION" environment variable\n'
	actual="$(set -a; HBLOCK_REDIRECTION='::1' runInTestShell "${SCRIPT_DIR:?}/../../hblock" -qO-)"
	expected="$(cat -- "${0%.sh}".out)"
	if ! assertEquals "${actual?}" "${expected?}"; then
		exit 1
	fi
}

main "${@-}"
