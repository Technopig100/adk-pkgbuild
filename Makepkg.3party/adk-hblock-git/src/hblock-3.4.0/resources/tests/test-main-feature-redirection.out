# Generated with hBlock 0.0.0 (https://example.com)
# Blocked domains: 26

# BEGIN HEADER
127.0.0.1       localhost hblock
255.255.255.255 broadcasthost
::1             localhost hblock
::1             ip6-localhost ip6-loopback
fe00::0         ip6-localnet
ff00::0         ip6-mcastprefix
ff02::1         ip6-allnodes
ff02::2         ip6-allrouters
ff02::3         ip6-allhosts
# END HEADER

# BEGIN BLOCKLIST
::1 entry-with-comment-000.com
::1 entry-with-comment-001.com
::1 entry-with-comment-002.com
::1 entry-with-comment-003.com
::1 entry-with-comment-004.com
::1 entry-with-prefix-000.com
::1 entry-with-prefix-001.com
::1 entry-with-prefix-002.com
::1 entry-with-prefix-003.com
::1 entry-with-prefix-004.com
::1 entry-with-prefix-005.com
::1 entry-with-prefix-006.com
::1 hblock-check.molinero.dev
::1 single-entry-000.com
::1 single-entry-001.com
::1 single-entry-002.com
::1 single-entry-003.com
::1 single-entry-017.com
::1 sub-000.single-entry-017.com
::1 sub-000.single-entry-018.com
::1 sub-000.sub-000.single-entry-017.com
::1 sub-000.sub-000.single-entry-018.com
::1 sub-000.sub-000.sub-000.single-entry-017.com
::1 sub-000.sub-000.sub-000.single-entry-019.com
::1 sub-001.single-entry-017.com
::1 sub-001.single-entry-018.com
# END BLOCKLIST
